packages/hooks/src/useCreateNFT.ts

```diff
@@ -184,22 +196,39 @@ export default function useCreateNFT(
           isAnimation,
           isPrivate,
         })
+        const metadata: MetadataInput = {
+          name,
+          image: media.image,
+          description,
+          animationUrl: media.animationUrl,
+          unlockableContent: media.unlockableContent,
+          attributes: (traits || []).map((x) => ({
+            traitType: x.type,
+            value: x.value,
+          })),
+        }
 
         // lazy minting
         if ((await config).hasLazyMint) {
-          const assetToCreate = {
-            standard,
+          const assetToCreate: LazyMintedAssetSignatureInput = {
             creatorAddress: account.toLowerCase(),
-            description,
-            name,
-            traits: traits || null,
-            ...media,
-            supply: amount ? amount.toString() : '1',
             royalties: royalties ? Math.round(royalties * 100) : null,
+            supply: amount ? amount.toString() : '1',
+            metadata: metadata,
+            // deprecated fields that are meant to be removed
+            standard: null,
+            animationUrl: null,
+            description: null,
+            image: null,
+            name: null,
+            traits: null,
+            unlockableContent: null,
```

Comment:
I think having null value will be rejected by API when those deprecated fields are removed from API.
It's a codegen "issue" to force optional field to be set. Maybe it's time to remove this limitation to limit future breaking changes of the lib.

Reply:
I can force the type because, yes, I agree this is ugly, forcing the type is not much better, but anyway, this will be gone soon.

Reply:
done [7cb2830](https://github.com/liteflow-labs/liteflow-js/pull/102/commits/7cb283024d9aa947a5b5aed44f54c879cb0a6c21)